x-pack/plugin/esql/qa/security/src/javaRestTest/java/org/elasticsearch/xpack/esql/EsqlSecurityIT.java

```diff
@@ -271,11 +271,32 @@ private void removeEnrichPolicy() throws Exception {
         client().performRequest(new Request("DELETE", "_enrich/policy/songs"));
     }
 
-    private Response runESQLCommand(String user, String command) throws IOException {
+    protected Response runESQLCommand(String user, String command) throws IOException {
         if (command.toLowerCase(Locale.ROOT).contains("limit") == false) {
             // add a (high) limit to avoid warnings on default limit
             command += " | limit 10000000";
         }
+        XContentBuilder json = JsonXContent.contentBuilder();
+        json.startObject();
+        json.field("query", command);
+        addRandomPragmas(json);
+        json.endObject();
+        Request request = new Request("POST", "_query");
+        request.setJsonEntity(Strings.toString(json));
+        request.setOptions(RequestOptions.DEFAULT.toBuilder().addHeader("es-security-runas-user", user));
+        return client().performRequest(request);
+    }
+
+    static void addRandomPragmas(XContentBuilder builder) throws IOException {
+        Settings pragmas = randomPragmas();
+        if (pragmas != Settings.EMPTY) {
```

Comment:
I imagine we have to skip this if we're running in a release build.

Reply:
Oh yes. I inadvertently removed this. I’ll add it back. Thanks.